### PR TITLE
[STORM-942] Add FluxParser method parseInputStream() to eliminate disk usage.

### PR DESCRIPTION
**Pull request by lyonbrw**

### CODE REVIEW

**Comment by lyonbrw** [comment]
Included reusable parseInputStream for use by parseFile and parseResource. Also allows a path for programatically creating topologies without the need to write to / read from disk.

File: external/flux/flux-core/src/main/java/org/apache/storm/flux/parser/FluxParser.java [hunk 1]
```diff
@@ -44,40 +44,47 @@ private FluxParser(){}
...
+        
+        return topology;
     }
+    
+    public static TopologyDef parseInputStream(InputStream inputStream, boolean dumpYaml, boolean processIncludes,
+            String propertiesFile, boolean envSub) throws IOException {
+		
+    	Yaml yaml = yaml();
+    	
+		if (inputStream == null) {
```
Comment (HeartSaVioR) [inline external/flux/flux-core/src/main/java/org/apache/storm/flux/parser/FluxParser.java]: @lyonbrw Indentation looks broken.

File: external/flux/flux-core/src/main/java/org/apache/storm/flux/parser/FluxParser.java [hunk 2]
```diff
@@ -44,40 +44,47 @@ private FluxParser(){}
...
+			LOG.error("Unable to load input stream");
+			System.exit(1);
+		}
+		
+		TopologyDef topology = loadYaml(yaml, inputStream, propertiesFile, envSub);
+		inputStream.close();
```
Comment (HeartSaVioR) [inline external/flux/flux-core/src/main/java/org/apache/storm/flux/parser/FluxParser.java]: @lyonbrw 
IMO it isn't a good way to close InputStream which is passed to a parameter. Caller can take care of it, and parseFile / parseResource already do it.

**Comment by lyonbrw** [comment]
Just saw that redundant close() function before you commented. My bad. ;) Thanks. Corrected both, build in progress.

**Comment by HeartSaVioR** [comment]
@lyonbrw 
Indentation is still broken. Please compare parseInputStream()'s indent level to other methods.

Btw, could you get rid of merge commit from PR if you don't mind?
It doesn't affect the source code of Storm, but leaves unrelated commit log.

**Comment by lyonbrw** [comment]
Apologize for the merge commit. What is the proper way to remove it?

**Comment by lyonbrw** [comment]
Indentation fixed.

**Comment by HeartSaVioR** [comment]
@lyonbrw 
Actually I recommended you to make new branch before do some works.
I can cherry-pick your commits into master, so let me handle it for now.

**Comment by HeartSaVioR** [comment]
OK, confirmed that all things fixed. +1.

**Comment by HeartSaVioR** [comment]
@lyonbrw 
I cherry-picked your efforts into master now. Would you close this PR?

**Comment by lyonbrw** [comment]
Will do. Thanks.

**Pull request closed without merging**